src/connectors/argent/argentMobile/modal/argentModal.ts

```diff
@@ -67,21 +92,66 @@ class ArgentModal {
   private overlay?: HTMLDivElement
   private popupWindow?: Window
   private closingTimeout?: NodeJS.Timeout
+  private standaloneConnectorModal?: Modal
+
+  public showWalletConnectModal(
+    wcUri: string,
+    modalWallet: ModalWalletExtended,
+  ) {
+    const wcParam = encodeURIComponent(wcUri)
+    const href = encodeURIComponent(window.location.href)
+
+    this.showModal(
+      {
+        desktop: `${this.bridgeUrl}?wc=${wcParam}&href=${href}&device=desktop&onlyQR=true`,
+        ios: `${this.mobileUrl}app/wc?uri=${wcParam}&href=${href}&device=mobile`,
+        android: `${this.mobileUrl}app/wc?uri=${wcParam}&href=${href}&device=mobile`,
+      },
+      modalWallet,
+    )
+  }
 
-  public showWalletConnectModal(wcUri: string) {
+  public getWalletConnectQR(wcUri: string) {
     const wcParam = encodeURIComponent(wcUri)
     const href = encodeURIComponent(window.location.href)
 
-    this.showModal({
-      desktop: `${this.bridgeUrl}?wc=${wcParam}&href=${href}&device=desktop`,
+    this.getQR({
+      desktop: `${this.bridgeUrl}?wc=${wcParam}&href=${href}&device=desktop&onlyQR=true`,
       ios: `${this.mobileUrl}app/wc?uri=${wcParam}&href=${href}&device=mobile`,
       android: `${this.mobileUrl}app/wc?uri=${wcParam}&href=${href}&device=mobile`,
     })
   }
 
+  private getQR(urls: Urls) {
+    const overlay = document.createElement("div")
+    const shadow = document.querySelector("#starknetkit-modal-container")
+
+    if (shadow?.shadowRoot) {
+      const slot = shadow.shadowRoot.querySelector(".qr-code-slot")
+
+      if (slot) {
+        slot.innerHTML = overlayHtmlOnlyQR
+        document.body.appendChild(overlay)
+        this.overlay = overlay
+
+        const iframe = slot.querySelector("iframe") as HTMLIFrameElement
+        iframe.setAttribute("src", urls.desktop)
+
+        for (const [key, value] of Object.entries(iframeStyleOnlyQR)) {
+          iframe.style[key as any] = value
+        }
+      } else {
+        throw new Error("Cannot find QR code slot")
+      }
+    } else {
+      throw new Error("Cannot find modal")
+    }
+  }
+
+  // TODO handle this
   public showApprovalModal(_: RequestArguments): void {
     if (device === "desktop") {
-      this.showModal({
+      this.showModalOld({
```

Comment:
why keeping the `showModalOld`?
saw a comment below, still needed or will just be removed in a followup pr?